### PR TITLE
Fixes to the group saving logic

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

File: mod/groups/actions/groups/edit.php [hunk 1]
```diff
@@ -116,7 +116,9 @@ function profile_array_decoder(&$v) {
 	// verify new owner is member and old owner/admin is logged in
 	if ($group->isMember(get_user($new_owner_guid)) && ($old_owner_guid == $user->guid || $user->isAdmin())) {
 		$group->owner_guid = $new_owner_guid;
-		$group->container_guid = $new_owner_guid;
+		if ($group->container_guid == $old_owner_guid) {
```
Comment (mrclay) [inline mod/groups/actions/groups/edit.php]: Could you add a comment for why this condition is necessary?

File: engine/classes/ElggGroup.php [hunk 2]
```diff
@@ -523,6 +523,12 @@ protected function create() {
 		global $CONFIG;
 		
 		$guid = parent::create();
+		if (!$guid) {
+			// @todo this probably means permission to create entity was denied
+			// Is returning false the correct thing to do
```
Comment (mrclay) [inline engine/classes/ElggGroup.php]: Seems right to me.

Comment (hypeJunction) [inline engine/classes/ElggGroup.php]: I just copied this from ElggObject

File: mod/groups/languages/en.php [hunk 3]
```diff
@@ -58,6 +58,7 @@
 	'groups:cantcreate' => 'You can not create a group. Only admins can.',
 	'groups:cantedit' => 'You can not edit this group',
 	'groups:saved' => 'Group saved',
+	'grupss:save_error' => 'Group could not be saved',
```
Comment (mrclay) [inline mod/groups/languages/en.php]: Typo in key.

File: engine/classes/ElggGroup.php (hunk 2 above)
Comment (hypeJunction) [inline engine/classes/ElggGroup.php]: I was having problems with saving a group and getting a Fatal Error. Due to permissions logic in another ticket, the group was never being saved, and this kept on trying to write to DB.

File: mod/groups/languages/en.php (hunk 3 above)
Comment (hypeJunction) [inline mod/groups/languages/en.php]: Oops, that's what you get for working at 5am

**Comment by hypeJunction** [comment]
@mrclay, see if this works.

File: engine/classes/ElggGroup.php (hunk 2 above)
Comment (mrclay) [inline engine/classes/ElggGroup.php]: I'm just going to answer the question "yes", now we don't need the comment in either place :)

File: mod/groups/actions/groups/edit.php (hunk 1 above)
Comment (mrclay) [inline mod/groups/actions/groups/edit.php]: Great

**Comment by mrclay** [comment]
LGTM, but would like another opinion

**Comment by jdalsem** [comment]
LGTM

**Pull request merged**